front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PipelineController.groovy

```diff
@@ -115,6 +115,23 @@ class PipelineController {
     pipelineDAO.update(pipelineId, pipeline)
   }
 
+  @RequestMapping(value = "/{id}", method = RequestMethod.PUT)
+  Pipeline update(@PathVariable String id, @RequestBody Pipeline pipeline) {
+    Pipeline existingPipeline = pipelineDAO.findById(id)
+    if (!existingPipeline || pipeline.id != id) {
+      throw new InvalidPipelineRequestException()
```

Comment:
If the existing pipeline doesn't exist, should that be a 404? (maybe `findById()` already throws a 404?)